src/XrdMacaroons/XrdMacaroonsAuthz.cc

```diff
@@ -504,18 +505,40 @@ AuthzCheck::verify_path(const unsigned char * pred, size_t pred_sz)
         return 1;
     }
 
-    int result = strncmp(pred_str.c_str(), m_path.c_str(), pred_str.size());
+    // Allow operations under subdirectories and not substrings 
+    // For e.g. pred_str = "/data/sudir/mydir"
+    // Allows m_path = /data/subdir/mydir/newdir
+    // But rejects, m_path = /data/subdir/mydirmycoolname/newdir
+    int result = !is_subdirectory(pred_str, m_path);
     if (!result)
```

Comment:
Why `result = !...` and `!result` in the condition? I guess we can use `int is_subdir = is_subdirectory(...)`, then `if (is_subdir) ...` to be clearer.

Reply:
Was complying with the existing structure as much as possible. Makes sense to change to `is_subdir`. Done